src/shared/ui-kit/components/TermsConditionModal/TermsConditionModal.tsx

```diff
@@ -0,0 +1,178 @@
+import { Button, Flex, Modal, ModalBody, ModalContent, ModalHeader, ModalOverlay } from "@chakra-ui/react";
+import { Color, Text } from "@shared/ui-kit";
+import { LocalStorageKeys } from "../constants";
+import { useState } from "react";
+
+export function TermsConditionModal() {
+  const localStorageState = localStorage.getItem(LocalStorageKeys.ACCEPTED_TERMS_AND_CONDITIONS);
+  const [acceptedCondition, setAcceptedCondition] = useState(localStorageState ? JSON.parse(localStorageState) : false);
+
```

Comment:
We can also use `import { useLocalStorage } from "@dex/hooks";` hook to manage the state for us instead handling it in component. like this 
`  const { value: termsAccepted, handleBooleanChange } = useLocalStorage<boolean | undefined>(
    LocalStorageKeys.ACCEPTED_TERMS_AND_CONDITIONS,
    false // default value
  );`
  
  Make a new export function in `useLocalStorageHook` like `function handleBooleanChange() {
    setValue(true as T);
  }` which will be called when user presses on `Accept Button`

Reply:
1 - `useLocalStorageHook` => `useLocalStorage`
2 - I would suggest naming `handleBooleanChange` to `setStorageValue`
3 - Setting the passed-in `true` value to the generic `T` in `setValue` opens up room for error since `true` is always a `boolean` but a developer can set the generic type T in the `useLocalStorage` hook to any type. It would be more accurate to either not type the boolean value, e.g. `setValue(true)`, or allow the developer to pass in a value to `setStorageKey`, .e.g 
```
function setStorageValue(newValue: T) {
   setValue(newValue);
}
```
The second suggestion is more dynamic.

Reply:
Done.